src/System.Text.Json/tests/WritableJsonApiTests.JsonNumberUnitTests.cs

```diff
@@ -0,0 +1,519 @@
+        {
+            var jsonNumber = new JsonNumber();
+            jsonNumber.SetUInt32(value);
+            uint result = jsonNumber.GetUInt32();
+            Assert.Equal(value, result);
+
+            bool success = jsonNumber.TryGetUInt32(out result);
+            Assert.True(success);
+            Assert.Equal(value, result);
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(17)]
+        [InlineData(ulong.MaxValue)]
+        public static void TestUInt64(ulong value)
+        {
+            var jsonNumber = new JsonNumber();
+            jsonNumber.SetUInt64(value);
+            ulong result = jsonNumber.GetUInt64();
+            Assert.Equal(value, result);
+
+            bool success = jsonNumber.TryGetUInt64(out result);
+            Assert.True(success);
+            Assert.Equal(value, result);
+        }
+
+        [Theory]
+        [InlineData("3.14")]
+        [InlineData("-17")]
+        [InlineData("0")]
+        [InlineData("189")]
+        [InlineData("1e400")]
+        [InlineData("1e+100000002")]
+        [InlineData("184467440737095516150.184467440737095516150")]
+        public static void TestString(string value)
+        {
+            var jsonNumber = new JsonNumber();
+            jsonNumber.SetFormattedValue(value);
+            string result = jsonNumber.ToString();
+            Assert.Equal(value, result);
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("3,14")]
+        [InlineData("this is not a number")]
+        [InlineData("NAN")]
+        [InlineData("0.")]
+        [InlineData("008")]
+        [InlineData("0e")]
+        [InlineData("5e")]
+        [InlineData("5a")]
+        [InlineData("0.1e")]
+        [InlineData("-01")]
+        [InlineData("10.5e")]
+        [InlineData("10.5e-")]
+        [InlineData("10.5e+")]
+        [InlineData("10.5e-0.2")]
+        [InlineData(" 6")]
+        [InlineData("6 ")]
+        [InlineData(" 6 ")]
+        [InlineData("+0")]
+        [InlineData("+1")]
+        public static void TestInvalidString(string value)
+        {
+            Assert.Throws<ArgumentException>(() => new JsonNumber(value));
+        }
+
+        [Fact]
+        public static void TestNullString()
+        {
+            Assert.Throws<NullReferenceException>(() => new JsonNumber(null));
+        }
+
+        [Fact]
+        public static void TestIntFromString()
+        {
+            var jsonNumber = new JsonNumber("145");
+            int result = jsonNumber.GetInt32();
+            Assert.Equal(145, result);
+        }
+
+        [Fact]
+        public static void TestUpcasts()
+        {
+            byte value = 17;
+            var jsonNumber = new JsonNumber(value);
+
+            // Geeting other types should also succeed:
+            short shortResult = jsonNumber.GetInt16();
+            Assert.Equal(value, shortResult);
+            bool success = jsonNumber.TryGetInt16(out shortResult);
```

Comment:
nit: It doesn't look like the `success` local is used for anything meaningful.

How about just assert:
```C#
Assert.True(jsonNumber.TryGetInt16(out shortResult));
```

Reply:
Sure, thanks!